### PR TITLE
[Bug] Fix process instance display and npe when save

### PR DESCRIPTION
**Pull request by 106umao**
When workflow definitions are removed, their associated workflow instances should be displayed

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java [hunk 1]
```diff
@@ -533,6 +533,10 @@ public Map<String, Object> updateProcessInstance(User loginUser, long projectCod
         }
         ProcessDefinition processDefinition =
                 processDefineMapper.queryByCode(processInstance.getProcessDefinitionCode());
+        if (processDefinition == null) {
+            putMsg(result, Status.PROCESS_DEFINE_NOT_EXIST,processInstance.getProcessDefinitionCode());
+            return result;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java:559]:
```suggestion
           throw new ServiceException(Status.PROCESS_DEFINE_NOT_EXIST,processInstance.getProcessDefinitionCode());
```

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapperTest.java [hunk 2]
```diff
@@ -45,6 +46,9 @@ public class ProcessInstanceMapperTest extends BaseDaoTest {
     @Autowired
     private ProcessDefinitionMapper processDefinitionMapper;
 
+    @Autowired
+    ProcessDefinitionLogMapper processDefinitionLogMapper;
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapperTest.java:50]:
```suggestion
    private ProcessDefinitionLogMapper processDefinitionLogMapper;
```

**Comment by 106umao** [comment]
> 

will do it

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql [hunk 3]
```diff
@@ -441,28 +441,31 @@ CREATE TABLE `t_ds_process_definition` (
...
+    `timeout`          int(11) DEFAULT '0' COMMENT 'time out,unit: minute',
+    `tenant_id`        int(11) NOT NULL DEFAULT '-1' COMMENT 'tenant id',
+    `execution_type`   tinyint(4) DEFAULT '0' COMMENT 'execution_type 0:parallel,1:serial wait,2:serial discard,3:serial priority',
+    `operator`         int(11) DEFAULT NULL COMMENT 'operator user id',
+    `operate_time`     datetime     DEFAULT NULL COMMENT 'operate time',
+    `create_time`      datetime NOT NULL COMMENT 'create time',
+    `update_time`      datetime NOT NULL COMMENT 'update time',
+    PRIMARY KEY (`id`)
+    UNIQUE KEY `log_unique` (`code`,`version`);
+    KEY `project_code_idx` (`project_code`)
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql:467]:
```suggestion
    PRIMARY KEY (`id`),
    UNIQUE KEY `log_unique` (`code`,`version`),
    KEY `project_code_idx` (`project_code`)
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Pull request closed without merging**